### ISSUE TITLE
Per-tenant OAuth 2.0 Settings | Django Multi-Tenant OAuth2 Credentials

### ISSUE THREAD

**Issue by duplxey** [wagnerdelima/drf-social-oauth2#252]
I have a multi-tenant Django app created using the [django-tenants](https://django-tenants.readthedocs.io/en/latest/) package. I successfully setup "global" social authentication using [`social_core.backends.google.GoogleOAuth2`](https://python-social-auth.readthedocs.io/en/latest/backends/google.html) backend.

Now, I'd like to allow tenants to provide their own Google OAuth2 credentials (`client_id` & `client_secret`) and authenticate through their own Google apps. Is there any way I can inject tenants' OAuth2 credentials in [`ConvertTokenView()`](https://github.com/wagnerdelima/drf-social-oauth2/blob/master/drf_social_oauth2/views.py#L115) based on what schema the request is made from?

Pseudocode:

```python
class ConvertTokenView(request):

  def post(self, request):
    tenant = get tenant based on schema
    backend.set_client_id(tenant.google_client_id)
    backend.set_client_secret(tenant.google_client_secret)
    backend.authenticate(request.body)
  
    return tokenResponse
```

Alternatively, I was thinking about dynamically modifying the following two *settings.py* settings:

```python
# these are global, but i'd like each tenant to have their own creds
SOCIAL_AUTH_GOOGLE_OAUTH2_KEY = "<google_client_id>"
SOCIAL_AUTH_GOOGLE_OAUTH2_SECRET = "<google_client_secret>"
```

However, this method does not work, because these two settings are only read once on Django server startup.

> I posted a similar question on [StackOverflow](https://stackoverflow.com/questions/79079854/django-enable-multi-tenant-social-login-tenants-provide-their-own-oauth2-creden).

**Comment by ex8** [comment]
I have tried this and this is not possible without forking this library and adding the functionality in yourself. I would advise implementing your own auth solution where you store the client id's and secrets in an encrypted vault. Essentially, you would need to implement the solution yourself.

**Comment by duplxey** [comment]
@ex8 Yeah, I ended up implementing my own OAuth2 flow and it seems to be working well.
